### PR TITLE
[Smartlab]Update smartlab models

### PR DESCRIPTION
**Pull request by wdkwyf**
mstcn++ -> smartlab-sequence-modelling-0002
Object detection models may have updated accuracy, wait for feedback now.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1 +1,2 @@
 *.pyc
+.idea/
```
Comment (Wovchena) [inline .gitignore:2]: OMZ itself doesn't produce this folder. This should be in your personal global .gitignore

File: models/intel/smartlab-sequence-modelling-0001/README.md [hunk 2]
```diff
@@ -1,153 +1,33 @@
 # smartlab-sequence-modelling-0001
 
 ## Use Case and High-Level Description
-This is an online action segmentation network for 16 classes trained on Intel dataset. It is an online version of MSTCN++. The difference between online MSTCN++ and MSTCN++ is that the former accept stream video as input while the latter assume the whole video is given.
-
-For the original MSTCN++ model details see [paper](https://arxiv.org/abs/2006.09220)
+This is a feature extractor that is based on Mobilenet-v3 network without origianl classifier layer. Input is RGB image and output is feature vector.
```
Comment (Wovchena) [inline models/intel/smartlab-sequence-modelling-0001/README.md:4]: Please, update https://github.com/openvinotoolkit/open_model_zoo/blob/5af689ea42c41581e8ae3d9f50a5eb0da60b5675/models/intel/device_support.md table

**Comment by wdkwyf** [comment]
hi, @Wovchena Could you please review? I didn't include the mobilenet-v3-small model accuracy checker config because I think in smartlab we only fine-tune the model and didn't change a lot. 
Now:
- smartlab-sequence-modelling-0001
- smartlab-action-recognition-0001-encoder-top
- smartlab-action-recognition-0001-encoder-side
- smartlab-action-recognition-0001-decoder

don't have AC config for the same reason.

**Comment by Wovchena** [comment]
I can see `smartlab-sequence-modelling-0001` is completely missing. That will make it impossible to download the model with omz_downloader when the model will be published.

I don't think it's a good idea to drop accuracy checker config. I guess it can be used for internal validation. @eaidova to comment

**Comment by eaidova** [comment]
> I don't think it's a good idea to drop accuracy checker config. I guess it can be used for internal validation. @eaidova to comment

OMZ is public open source repo and should contains actual content for users, if we do not have model in repo, we should not store configs here, because users can not try it, we should not care about internal stuff (it managed in different way)

**Comment by wdkwyf** [comment]
> > I don't think it's a good idea to drop accuracy checker config. I guess it can be used for internal validation. @eaidova to comment
> 
> OMZ is public open source repo and should contains actual content for users, if we do not have model in repo, we should not store configs here, because users can not try it, we should not care about internal stuff (it is managed in different way)

hi, the meaning is we must put AC config in the every model directory?

**Comment by eaidova** [comment]
> hi, the meaning is we must put AC config in the every model directory? Can I only put readme and auto-generated model.yml files?

if you can. yes, if not, we will provide them from our side

**Comment by wdkwyf** [comment]
> > hi, the meaning is we must put AC config in the every model directory? Can I only put readme and auto-generated model.yml files?
> 
> if you can. yes, if not, we will provide them from our side

Hi, I can finish the mobilenet+mstcn part AC, and need your help to finish the mobilenet+TSM part. Thanks!

**Comment by wdkwyf** [comment]
hi, @Wovchena @eaidova . Could you please review this pr?

**Comment by Wovchena** [comment]
> It's kind of strange why these following models don't have directory here but still stored in https://[storage.openvinotoolkit.org](https://storage.openvinotoolkit.org/repositories/open_model_zoo/2022.1/models_bin/2/smartlab-action-recognition-0001/)/repositories/open_model_zoo/2022.1/models_bin/2/smartlab-action-recognition-0001/
> 
> * smartlab-action-recognition-0001-encoder-top
> * smartlab-action-recognition-0001-encoder-side
> * smartlab-action-recognition-0001-decoder

I don't know what was happening here. 

Can you add `martlab-action-recognition-0001` folder with `README.md`? Just for your reference, here is the future layout with `smartlab-action-recognition-0001`: https://github.com/openvinotoolkit/open_model_zoo/pull/3572

**Comment by wdkwyf** [comment]
Hi, @Wovchena @eaidova . I finished smartlab-action-recognition-0001 related readme commit.

File: models/intel/index.md [hunk 3]
```diff
@@ -475,8 +475,11 @@ Deep Learning models for online sequence modeling.
 
 | Model Name | Complexity (GFLOPs) | Size (Mp) |
 |------------|---------------------|-----------|
-| [smartlab-sequence-modelling-0001](./smartlab-sequence-modelling-0001/README.md) | 0.049   | 1.02    |
-
+| [smartlab-sequence-modelling-0001](./smartlab-sequence-modelling-0001/README.md) | 0.11   | 2.537    |
+| [smartlab-sequence-modelling-0002](./smartlab-sequence-modelling-0002/README.md) | 0.049   | 1.02    |
+| [smartlab-action-recognition-0001](./smartlab-action-recognition-0001/README.md) |    -   |   -      |
+|   smartlab-action-recognition-encoder-0001                                       | 0.611   | 3.387   |
+|   smartlab-action-recognition-decoder-0001                                       | 0.008   | 4.099   |
```
Comment (Wovchena) [inline models/intel/index.md:482]: I think there should be three `smartlab-action-recognition` models:
`smartlab-action-recognition-0001-decoder`
`smartlab-action-recognition-0001-encoder-side`
`smartlab-action-recognition-0001-encoder-top`

File: models/intel/smartlab-action-recognition-0001/smartlab-action-recognition-0001-decoder/README.md [hunk 4]
```diff
@@ -0,0 +1,34 @@
+# smartlab-action-recognition-0001 (decoder)
```
Comment (Wovchena) [inline models/intel/smartlab-action-recognition-0001/smartlab-action-recognition-0001-decoder/README.md:1]: We usually keep only one readme for composite models. For example https://github.com/openvinotoolkit/open_model_zoo/tree/master/models/intel/text-spotting-0005

**Review by Wovchena — CHANGES_REQUESTED** [review]

File: models/intel/smartlab-action-recognition-0001/smartlab-action-recognition-0001-decoder/README.md (hunk 4 above)
Comment (wdkwyf) [reply models/intel/smartlab-action-recognition-0001/smartlab-action-recognition-0001-decoder/README.md:1]: @Wovchena If I didn't separate readme, I will add .gitkeep file in these empty directories. Is it OK?

File: models/intel/smartlab-action-recognition-0001/README.md [hunk 5]
```diff
@@ -10,12 +10,65 @@ These models are fine-tuned with smartlab dataset to predict actions and can cla
...
+| ---------------------------------------------- | ------------------ |
+| Accuracy on the DSI1867                        | TODO               |
+| Source framework                               | PyTorch\*          |
 
-Detailed explaination is available in [smartlab-action-recognition-0001-decoder](./smartlab-action-recognition-0001-decoder/README.md)
+## Encoder model specification
 
+The smartlab-action-recognition-encoder-0001 is a Mobilenet-V2 like backbone with convolutional encoder part of the action recognition. 
+
+To use both top-view and side-view features, there are two similar models called: `smartlab-action-recognition-0001-encoder-side` and `smartlab-action-recognition-0001-encoder-top`, which have the same strcuture but different weights.
```
Comment (Wovchena) [inline models/intel/smartlab-action-recognition-0001/README.md:25]:
```suggestion
## Encoder models specification

The smartlab-action-recognition-0001-encoder-* are have Mobilenet-V2 like backbone with convolutional encoder part of the action recognition. 

There are two models called: `smartlab-action-recognition-0001-encoder-side` and `smartlab-action-recognition-0001-encoder-top`, which have the same strcuture but different weights.
```

File: models/intel/smartlab-action-recognition-0001/README.md [hunk 6]
```diff
@@ -10,12 +10,65 @@ These models are fine-tuned with smartlab dataset to predict actions and can cla
...
+- `W` - image width
+Expected color order is `BGR`
+
+### Outputs
+
+1. Name: `output_feature`, shape: `1, 1280`. Features from encoder part of action recogntion head.
+
+## Decoder model specification
+
+The smartlab-action-recognition-decoder-0001 is a fully connected decoder part which accepts features from top and front views, computed by encoder and predicts score for action across following label list: `no_action`, `noise_action`,  `adjust_rider`
```
Comment (Wovchena) [inline models/intel/smartlab-action-recognition-0001/README.md:48]:
```suggestion
The smartlab-action-recognition-0001-decoder is a fully connected decoder part which accepts features from top and front views, computed by encoder and predicts score for action across following label list: `no_action`, `noise_action`,  `adjust_rider`
```

File: models/intel/smartlab-action-recognition-0001/smartlab-action-recognition-0001-decoder/README.md (hunk 4 above)
Comment (Wovchena) [reply models/intel/smartlab-action-recognition-0001/smartlab-action-recognition-0001-decoder/README.md]: Can you remove such folders instead? Merging https://github.com/openvinotoolkit/open_model_zoo/pull/3572 will recreate them.

Comment (wdkwyf) [reply models/intel/smartlab-action-recognition-0001/smartlab-action-recognition-0001-decoder/README.md]: Sure, removed.

**Review by Wovchena — APPROVED** [review]

**Review by eaidova — APPROVED** [review]

**Pull request merged**